docs/serializer-formatter.rst

```diff
@@ -72,62 +72,130 @@ to Kafka. A ``SerializationException`` may occur during the send call, if the da
-    KafkaAvroDecoder keyDecoder = new KafkaAvroDecoder(vProps);
-    KafkaAvroDecoder valueDecoder = new KafkaAvroDecoder(vProps);
-
-    ConsumerConnector consumer = kafka.consumer.Consumer.createJavaConsumerConnector(new ConsumerConfig(props));
-
-    Map<String, List<KafkaStream<Object, Object>>> consumerMap = consumer.createMessageStreams(
-        topicCountMap, keyDecoder, valueDecoder);
-    KafkaStream stream = consumerMap.get(topic).get(0);
-    ConsumerIterator it = stream.iterator();
-    while (it.hasNext()) {
-      MessageAndMetadata messageAndMetadata = it.next();
-      try {
-        String key = (String) messageAndMetadata.key();
-        IndexedRecord value = (IndexedRecord) messageAndMetadata.message();
-
-        ...
-      } catch(SerializationException e) {
-        // may need to do something with it
+    final Consumer<String, GenericRecord> consumer = new KafkaConsumer<String, String>(props);
+    consumer.subscribe(Arrays.asList(topic));
+
+    try {
+      while (true) {
+        ConsumerRecords<String, String> records = consumer.poll(100);
+        for (ConsumerRecord<String, String> record : records) {
+          System.out.printf("offset = %d, key = %s, value = %s \n", record.offset(), record.key(), record.value());
+        }
       }
+    } finally {
+      consumer.close();
     }
 
-We recommend users use the new producer in ``org.apache.kafka.clients.producer.KafkaProducer``. If
-you are using a version of Kafka older than 0.8.2.0, you can plug ``KafkaAvroEncoder`` into the old
-producer in ``kafka.javaapi.producer``. However, there will be some limitations. You can only use
-``KafkaAvroEncoder`` for serializing the value of the message and only send value of type Avro record.
-The Avro schema for the value will be registered under the subject *recordName-value*, where
-*recordName* is the name of the Avro record. Because of this, the same Avro record type shouldn't
-be used in more than one topic.
+
+Subject Name Strategy
+^^^^^^^^^^^^^^^^^^^^^
+
+By default, the KafkaAvroSerializer and KafkaAvroDeserializer default to using *<topicName>-Key*
+and *<topicName>-value* as the corresponding subject name while registering or retrieving the
+schema.
+
+This behavior can be modified by using the following configs
+
+``key.subject.name.strategy``
+  Determines how to construct the subject name under which the key schema is registered with the schema registry. By default, <topic>-key is used as subject.
+
+  * Type: class
+  * Default: class io.confluent.kafka.serializers.subject.TopicNameStrategy
+  * Importance: medium
+
+``value.subject.name.strategy``
+  Determines how to construct the subject name under which the value schema is registered with the schema registry. By default, <topic>-value is used as subject.
+
+  * Type: class
+  * Default: class io.confluent.kafka.serializers.subject.TopicNameStrategy
+  * Importance: medium
+
+The other available options that can be configured out of the box include
```

Comment:
Probably need to say something about any implementation of `io.confluent.kafka.serializers.subject.SubjectNameStrategy` can be specified.